p2p/switch.go

```diff
@@ -495,13 +520,46 @@ func (sw *Switch) IsDialingOrExistingAddress(addr *NetAddress) bool {
 		(!sw.config.AllowDuplicateIP && sw.peers.HasIP(addr.IP))
 }
 
+// AddPersistentPeers allows you to set persistent peers. It ignores
+// ErrNetAddressLookup. However, if there are other errors, first encounter is
+// returned.
+func (sw *Switch) AddPersistentPeers(addrs []string) error {
+	sw.Logger.Info("Adding persistent peers", "addrs", addrs)
+	netAddrs, errs := NewNetAddressStrings(addrs)
+	// report all the errors
+	for _, err := range errs {
+		sw.Logger.Error("Error in peer's address", "err", err)
+	}
+	// return first non-ErrNetAddressLookup error
+	for _, err := range errs {
+		if _, ok := err.(ErrNetAddressLookup); ok {
+			continue
```

Comment:
Doesn't this mean we might include `nil` in the list? What is the value of ignoring this error here if we don't even get a netAddr out of it?

Reply:
Good catch 👍 I'll need to think about it. See my other comment https://github.com/tendermint/tendermint/pull/3593#discussion_r280971995

Reply:
> Doesn't this mean we might include nil in the list?

no https://github.com/tendermint/tendermint/blob/develop/p2p/netaddress.go#L134-L138